OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java

```diff
@@ -1999,5 +2000,28 @@ private static double measuredDist(int x1, int y1, int x2, int y2) {
 				MapUtils.get31LatitudeY(y2), MapUtils.get31LongitudeX(x2));
 	}
 
+	private void avoidKeepForThroughMoving(List<RouteSegmentResult> result) {
+		for (int i = 1; i < result.size(); i++) {
+			RouteSegmentResult curr = result.get(i);
+			TurnType turnType = curr.getTurnType();
+			if (turnType == null) {
+				continue;
+			}
+			if (!turnType.keepLeft() && !turnType.keepRight()) {
+				continue;
+			}
+			RouteSegmentResult prev = result.get(i - 1);
+			String turnLanesPrev = getTurnLanesString(prev);
+			if (turnLanesPrev == null || !turnLanesPrev.contains("through")) {
```

Comment:
why do we check prev for through ?

Reply:
We check if present lanes tag in previous segment for to be sure that "through" instruction is not code generated, but from real data.